src/main/groovy/io/cify/framework/recording/RecordingController.groovy

```diff
@@ -142,16 +170,33 @@ class RecordingController {
     private static String getVideoDirForDevice(Device device) {
         return System.getProperty("videoDir") +
                 System.getProperty("task", "plug-and-play") +
-                "/" +
-                device.id +
-                "/"
+                "/" + device.id + TEMP
+    }
+
+    /**
+     * Gets output video path
+     * */
+    private static String getOutputVideoDirForDevice(Device device) {
+
+        String taskName = System.getProperty("task")
```

Comment:
if these values are empty maybe setting some default values would prevent a crash?

Reply:
videoDir default value is set to "build/cify/videos/"